scripts/india_census/common/base.py

```diff
@@ -0,0 +1,249 @@
+        if (row["Level"]).upper() == "INDIA":
+            return "0"
+        elif (row["Level"]).upper() == "STATE":
+            return row["State"]
+        elif (row["Level"]).upper() == "DISTRICT":
+            return row["District"]
+        elif (row["Level"]).upper() == "SUBDISTT":
+            return row["Subdistt"]
+        elif (row["Level"]).upper() == "TOWN":
+            return row["Town/Village"]
+        elif (row["Level"]).upper() == "VILLAGE":
+            return row["Town/Village"]
+        else:
+            raise Exception("Location Level not supported")
+
+    def _format_data(self):
+
+        #Generate Census locationid
+        self.raw_df["census_location_id"] = self.raw_df.apply(
+            self._format_location, axis=1)
+
+        #Remove the unwanted columns
+        #They are census codes which we dont use
+        #State,District,Subdistt,Town/Village,Ward,EB
+        self.raw_df.drop([
+            "State", "District", "Subdistt", "Town/Village", "Ward", "EB",
+            "Level", "Name"
+        ],
+                         axis=1,
+                         inplace=True)
+        #first column is Name of the place
+        #second column is Name of the TRU/placeOfResidence
+        #3-N are the actual values
+        value_columns = list(self.raw_df.columns[1:-1])
+
+        #converting rows in to columns. So the filnal structure will be
+        #Name,TRU,columnName,value
+        self.raw_df = self.raw_df.melt(id_vars=["census_location_id", "TRU"],
+                                       value_vars=value_columns,
+                                       var_name='columnName',
+                                       value_name='Value')
+
+        #Add corresponding StatisticalVariable, based on columnName and TRU
+        self.raw_df['StatisticalVariable'] = self.raw_df.apply(
+            lambda row: self.stat_var_index["{0}_{1}".format(
+                row["columnName"], row["TRU"])],
+            axis=1)
+        #add the census year
+        self.raw_df['Year'] = self.census_year
+
+        #Export it as CSV. It will have the following columns
+        #Name,TRU,columnName,value,StatisticalVariable,Year
+        self.raw_df.to_csv(self.csv_file_path, index=False, header=True)
+
+    def _get_base_name(self, row):
+        #This function is overridden in the child class
+        name = "Count_" + row["populationType"]
+        return name
+
+    def _get_base_constraints(self, row):
+        #This function is overridden in the child class
+        constraints = ""
+        return constraints
+
+    def _create_variable(self, data_row, place_of_residence=None):
+        row = copy.deepcopy(data_row)
+        name = self._get_base_name(row)
+        constraints = self._get_base_constraints(row)
+
+        if row["age"] == "YearsUpto6":
+            name = name + "_" + "YearsUpto6"
+            constraints = constraints + "age: dcid:YearsUpto6 \n"
```

Comment:
drop trailing space?

Or, could make name and constraints an array that you finally join with '_' and '\n' respectively.

Reply:
Done.